src/main.cpp

```diff
@@ -183,14 +183,22 @@ class AP4_DASHStream : public AP4_ByteStream
 Kodi Streams implementation
 ********************************************************/
 
-bool adaptive::AdaptiveTree::download(const char* url)
+bool adaptive::AdaptiveTree::download(const char* url, const std::map<std::string, std::string> &manifestHeaders)
 {
   // open the file
   void* file = xbmc->CURLCreate(url);
   if (!file)
     return false;
   xbmc->CURLAddOption(file, XFILE::CURL_OPTION_PROTOCOL, "seekable", "0");
   xbmc->CURLAddOption(file, XFILE::CURL_OPTION_PROTOCOL, "acceptencoding", "gzip");
+  if (!manifestHeaders.empty())
```

Comment:
Not needed to request for empty() here.

If you are lazy in writing you could also do:
for (const auto &entry :  manifestHeaders)
{